common/expression/src/evaluator.rs

```diff
@@ -92,17 +92,67 @@ impl Evaluator {
                 (Scalar::UInt8(val), DataType::UInt16) => {
                     Some(Value::Scalar(Scalar::UInt16(val as u16)))
                 }
+                (Scalar::UInt8(val), DataType::UInt32) => {
+                    Some(Value::Scalar(Scalar::UInt32(val as u32)))
+                }
+                (Scalar::UInt16(val), DataType::UInt32) => {
+                    Some(Value::Scalar(Scalar::UInt32(val as u32)))
+                }
+                (Scalar::UInt8(val), DataType::UInt64) => {
+                    Some(Value::Scalar(Scalar::UInt64(val as u64)))
+                }
+                (Scalar::UInt16(val), DataType::UInt64) => {
+                    Some(Value::Scalar(Scalar::UInt64(val as u64)))
+                }
+                (Scalar::UInt32(val), DataType::UInt64) => {
+                    Some(Value::Scalar(Scalar::UInt64(val as u64)))
+                }
                 (Scalar::Int8(val), DataType::Int16) => {
                     Some(Value::Scalar(Scalar::Int16(val as i16)))
                 }
                 (Scalar::UInt8(val), DataType::Int16) => {
                     Some(Value::Scalar(Scalar::Int16(val as i16)))
                 }
+                (Scalar::Int8(val), DataType::Int32) => {
+                    Some(Value::Scalar(Scalar::Int32(val as i32)))
+                }
+                (Scalar::Int16(val), DataType::Int32) => {
+                    Some(Value::Scalar(Scalar::Int32(val as i32)))
+                }
+                (Scalar::UInt8(val), DataType::Int32) => {
+                    Some(Value::Scalar(Scalar::Int32(val as i32)))
+                }
+                (Scalar::UInt16(val), DataType::Int32) => {
+                    Some(Value::Scalar(Scalar::Int32(val as i32)))
+                }
+                (Scalar::Int8(val), DataType::Int64) => {
+                    Some(Value::Scalar(Scalar::Int64(val as i64)))
+                }
+                (Scalar::Int16(val), DataType::Int64) => {
+                    Some(Value::Scalar(Scalar::Int64(val as i64)))
+                }
+                (Scalar::Int32(val), DataType::Int64) => {
+                    Some(Value::Scalar(Scalar::Int64(val as i64)))
+                }
+                (Scalar::UInt8(val), DataType::Int64) => {
+                    Some(Value::Scalar(Scalar::Int64(val as i64)))
+                }
+                (Scalar::UInt16(val), DataType::Int64) => {
+                    Some(Value::Scalar(Scalar::Int64(val as i64)))
+                }
+                (Scalar::UInt32(val), DataType::Int64) => {
```

Comment:
Should we consider `wrapping cast` from larger size integer to a shorter size integer?

Reply:
there will lose precision after downgrade cast 🤔

Reply:
We're still hesitant whether to throw an evaluation error or wrap the numbers on downgrade cast. Before consensus is met, I'm good to simply disable downgrade cast.